monitor/supply.py

```diff
@@ -0,0 +1,130 @@
+    now = maya.now()
+    vest_saft2_team_factor = vesting_remaining_factor(vesting_months=SAFT2_TEAM_VESTING_MONTHS, cliff=False, now=now)
+    vest_worklock_factor = vesting_remaining_factor(vesting_months=WORKLOCK_VESTING_MONTHS, cliff=True, now=now)
+    vest_nuco_factor = vesting_remaining_factor(vesting_months=NUCO_VESTING_MONTHS, cliff=True, now=now)
+    vest_university_factor = vesting_remaining_factor(vesting_months=UNIVERSITY_VESTING_MONTHS, cliff=True, now=now)
+    vested_nu = NU(0, 'NU')
+
+    saft2_locked_supply = NU(value=(SAFT2_INITIAL_SUPPLY.to_nunits() * vest_saft2_team_factor), denomination='NuNit')
+    vested_nu += (SAFT2_INITIAL_SUPPLY - saft2_locked_supply)
+
+    team_locked_supply = NU(value=(TEAM_INITIAL_SUPPLY.to_nunits() * vest_saft2_team_factor), denomination='NuNit')
+    vested_nu += (TEAM_INITIAL_SUPPLY - team_locked_supply)
+
+    nuco_locked_supply = NU(value=(NUCO_INITIAL_SUPPLY.to_nunits() * vest_nuco_factor), denomination='NuNit')
+    vested_nu += (NUCO_INITIAL_SUPPLY - nuco_locked_supply)
+
+    worklock_locked_supply = NU(value=(economics.worklock_supply * vest_worklock_factor), denomination='NuNit')
+    vested_nu += NU(value=(economics.worklock_supply - worklock_locked_supply.to_nunits()), denomination='NuNit')
+
+    university_locked_supply = NU(value=(UNIVERSITY_INITIAL_SUPPLY.to_nunits() * vest_university_factor),
+                                  denomination='NuNit')
+    vested_nu += (UNIVERSITY_INITIAL_SUPPLY - university_locked_supply)
+
+    locked_allocations['saft2'] = str(round(saft2_locked_supply, 2))
+    locked_allocations['team'] = str(round(team_locked_supply, 2))
+    locked_allocations['company'] = str(round(nuco_locked_supply, 2))
+    locked_allocations['worklock'] = str(round(worklock_locked_supply, 2))
+    locked_allocations['university'] = str(round(university_locked_supply, 2))
+
+    total_locked_allocations = (saft2_locked_supply + team_locked_supply + nuco_locked_supply +
+                                worklock_locked_supply + university_locked_supply)
+
+    # - Unlocked Allocations
+    unlocked_supply_info = OrderedDict()
+    initial_supply_info['unlocked_allocations'] = unlocked_supply_info
+    unlocked_supply_info['saft1'] = str(round(SAFT1_SUPPLY, 2))
+    unlocked_supply_info['casi'] = str(round(CASI_SUPPLY, 2))
+    unlocked_supply_info['vested'] = str(round(vested_nu, 2))
+    ecosystem_supply = INITIAL_SUPPLY - total_locked_allocations - (SAFT1_SUPPLY + CASI_SUPPLY + vested_nu)
+    unlocked_supply_info['ecosystem'] = str(round(ecosystem_supply, 2))
+
+    total_unlocked_allocations = SAFT1_SUPPLY + CASI_SUPPLY + vested_nu + ecosystem_supply
+
+    # Staking Rewards Information
+    staking_rewards_info = OrderedDict()
+    supply_info['staking_rewards_supply'] = staking_rewards_info
+    initial_supply_with_rewards = economics.initial_supply  # economics.initial_supply includes issued rewards
```

Comment:
```suggestion
    initial_supply_with_rewards = economics.token_supply_at_period  # economics.initial_supply includes issued rewards
```
Is this used to return the supply in this current moment (i.e. right now would be the initial 1bn + ~12m rewards issued since launch). If so the `BaseEconomics` class doesn't actually provide an expression for the supply in a given moment, but `StandardTokenEconomics` does

Reply:
Was working off of @KPrasch's point - https://discord.com/channels/411401661714792449/412681039178366996/770397121831501854 - that initial supply already includes rewards.

Reply:
@arjunhassard thanks for raising this - just realized that I should not create an Economics class in the constructor, and reuse it - rather, I should obtain a fresh copy of the  Economics every time the endpoint is invoked.

re: StandardTokenEconomics - I'm not sure I have direct access to that class.